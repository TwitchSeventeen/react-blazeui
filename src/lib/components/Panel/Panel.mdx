---
name: Panel
route: /objects/panel
menu: Objects
---

import { Playground, PropsTable } from 'docz'
import Panel from './index';

# Panel

A panel can be a very useful piece of layout, it provides a manageable scrollable space within the existing structure of your page.

## Basic usage
<Playground>
    <Panel style={{ height: '350px' }}>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam eget neque mollis, sodales nulla ut, porta
            urna. Vivamus gravida placerat metus ac malesuada. Donec convallis dolor in ex tristique, luctus tempus dui
            sollicitudin. Suspendisse enim urna, auctor ac viverra eget, venenatis non massa. Nulla magna est,
            ullamcorper eget feugiat ac, condimentum id diam. Cras tempor felis sit amet facilisis sollicitudin. Ut
            posuere malesuada feugiat. Morbi ex ex, pretium sit amet dolor eget, malesuada gravida est. Nullam id
            malesuada massa, ac facilisis elit. Nullam ut metus sit amet nisi finibus mollis. Praesent augue felis,
            feugiat placerat aliquam ut, ultrices sit amet dolor. Phasellus porta ligula nec sapien tempus commodo.
            Pellentesque hendrerit aliquet ornare. Integer nec tempus quam, vitae bibendum lectus. Suspendisse
            vestibulum pulvinar neque at scelerisque. Suspendisse accumsan dolor vitae nulla commodo, volutpat commodo
            nunc bibendum.
        
            Integer ut magna ac velit pharetra molestie. Nullam vestibulum est mattis, posuere augue id, suscipit nulla.
            Praesent vestibulum sapien hendrerit, sodales elit et, eleifend est. Aliquam ornare iaculis diam, commodo
            iaculis erat consectetur a. Quisque ut porta est. Mauris vulputate est eu auctor vestibulum. Vestibulum
            velit libero, cursus vel viverra et, fringilla et est. Donec nibh ipsum, finibus eu urna sed, ullamcorper
            sollicitudin velit. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae;
            Interdum et malesuada fames ac ante ipsum primis in faucibus. Nulla sit amet bibendum nunc, ut iaculis
            tellus. Aenean ornare velit vel venenatis scelerisque. Ut ac elit enim. Nulla nec congue ex. Cras nulla
            ligula, porttitor non pulvinar fermentum, hendrerit sit amet neque.
        
            Etiam pulvinar suscipit metus, at lobortis erat eleifend id. Suspendisse scelerisque sem et orci rhoncus
            posuere. Vivamus faucibus mi et feugiat elementum. Nulla tempor, ligula non ullamcorper condimentum, lorem
            lorem sagittis tortor, at interdum tellus nunc elementum purus. Nulla tincidunt augue nibh. Ut et augue a ex
            pretium sodales. Pellentesque quam neque, tincidunt eu fringilla nec, volutpat eget risus.
        
            Fusce sit amet arcu sodales, vestibulum quam vel, lacinia mauris. Mauris ut placerat libero, eu gravida
            lectus. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut lacinia felis vel nibh blandit, in
            vestibulum nisl congue. Integer ac dignissim nisi, id vehicula tellus. Class aptent taciti sociosqu ad
            litora torquent per conubia nostra, per inceptos himenaeos. In posuere vitae metus in fermentum.
        
            Quisque venenatis ut eros at posuere. Etiam pulvinar, ipsum eu interdum vestibulum, arcu urna dapibus lorem,
            quis volutpat tortor enim sed leo. Donec dapibus lacinia purus. Nullam ac porttitor ligula. Fusce aliquet
            mauris tortor, id euismod neque tempus vel. Fusce luctus ex et volutpat pretium. Nulla varius justo quis
            aliquam accumsan.
    </Panel>
</Playground>

## Properties
<PropsTable of={Panel} />