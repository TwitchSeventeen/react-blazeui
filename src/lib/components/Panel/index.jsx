import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Panel(props) {
  const { children, className, ...other } = props;
  const classes = classNames(
    'o-panel-container',
    className,
  );
  return (
    <div {...other} className={classes}>
      <div className="o-panel">
        { children }
      </div>
    </div>
  );
}

Panel.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
};

Panel.defaultProps = {
  children: null,
  className: null,
};

export default Panel;
