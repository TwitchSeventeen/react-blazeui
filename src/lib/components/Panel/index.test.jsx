import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Panel from './index';

describe('Panel', () => {
  it('should be defined', () => {
    expect(Panel).toBeDefined();
  });

  it('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <Panel style={{ height: '200px' }}>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Cupiditate harum iure molestias
        nihil quasi? Delectus impedit minus molestiae, possimus provident quidem vero. Ab
        dignissimos reprehenderit voluptatem. Assumenda commodi explicabo ullam!
      </Panel>,
    );

    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
