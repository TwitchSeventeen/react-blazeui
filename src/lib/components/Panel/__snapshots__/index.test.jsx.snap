// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel renders correctly 1`] = `
<div
  className="o-panel-container"
  style={
    Object {
      "height": "200px",
    }
  }
>
  <div
    className="o-panel"
  >
    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Cupiditate harum iure molestias nihil quasi? Delectus impedit minus molestiae, possimus provident quidem vero. Ab dignissimos reprehenderit voluptatem. Assumenda commodi explicabo ullam!
  </div>
</div>
`;
